clusterloader2/pkg/measurement/common/dns/dns_performance-k8s-hostnames.go

```diff
@@ -182,6 +221,35 @@ func (m *dnsPerfK8sHostnamesMeasurement) deleteDNSClientPermissions() error {
 	return m.k8sClient.RbacV1().ClusterRoleBindings().Delete(context.TODO(), dnsPerfTestPermissionsName, metav1.DeleteOptions{})
 }
 
+func (m *dnsPerfK8sHostnamesMeasurement) deleteDNSClient() error {
+	klog.Infof("Deleting deployment %q for measurement %q", dnsPerfTestDeploymentName, dnsPerfK8sHostnamesMeasureName)
+	if err := m.k8sClient.AppsV1().Deployments(m.testClientNamespace).Delete(context.TODO(), dnsPerfTestDeploymentName, metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+
+	options := &measurementutil.WaitForPodOptions{
+		DesiredPodCount:     func() int { return 0 },
+		CallerName:          m.String(),
+		WaitForPodsInterval: 10 * time.Second,
+	}
+
+	objectSelector := &util.ObjectSelector{
+		Namespace:     m.testClientNamespace,
+		LabelSelector: "dns-test=dnsperfgo",
+	}
+	podStore, err := measurementutil.NewPodStore(m.k8sClient, objectSelector)
+	if err != nil {
+		return fmt.Errorf("creating pod store during deletion: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 2*time.Minute)
+	defer cancel()
+	if _, err := measurementutil.WaitForPods(ctx, podStore, options); err != nil {
+		return fmt.Errorf("wating for DNS client pods to delete: %v", err)
+	}
+	return nil
+}
+
 func (m *dnsPerfK8sHostnamesMeasurement) cleanUp() error {
```

Comment:
don't you have to cleanup all the additional manifests that you have installed?

Reply:
I wasn't able to figure a good way to cleanup unstructured manifests. I can play around with invoking a `kubectl delete`

Reply:
There must be a way to parse a manifest file and issue a delete object request.
Try to see if you can use this https://github.com/rahulkjoshi/perf-tests/blob/master/clusterloader2/pkg/test/simple_test_executor.go#L331

Otherwise, `kubectl delete` would be ok.

Reply:
Added cleanup code to delete the additional manifests.